files/edge-core/scripts/edge-core-factory-reset

```diff
@@ -36,14 +35,7 @@ find /var/log -type f -print0 | xargs -0 truncate --size=0
 # edge-core handles the mbed credentials on factory reset and it
 # has different behavior based on whether it's configured for
 # dev mode or factory mode.
-if [ -e ${SNAP_DATA}/userdata/mbed ]; then
-    cp ${SNAP_DATA}/userdata/mbed ${SNAP_COMMON}/mbed-backup
-fi
-rm -rf ${SNAP_DATA}
+[ -n "${SNAP_DATA}" ] && find ${SNAP_DATA} -not -path "**/userdata/mbed/**" -exec rm -rf {}
```

Comment:
Need a semi-colon at the end of exec: `-exec rm -rf {} \;`
Notice it also has to be escaped so that bash won't interpret it.

Reply:
I've done some testing on this find command.  Here's what works for me:

```
[ -n "${SNAP_DATA}" ] && find ${SNAP_DATA} -depth -not -wholename "${SNAP_DATA}" -not -path "${SNAP_DATA}/userdata/mbed*" -exec rm -fd {} \;
```